tools/m2kcli/main.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 Analog Devices, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include <iostream>
+#include "commands/command.h"
+#include "commands/analog/analog_in.h"
+#include "commands/analog/analog_out.h"
+#include "commands/digital/digital.h"
+#include "commands/analog/power_supply.h"
+#include <libm2k/contextbuilder.hpp>
+
+static const char *const helpMessage = "Usage:\n"
+				       "m2kcli [-h | --help]\n"
+				       "       <command> [<args>]\n"
+				       "\n"
+				       "Control the ADALM2000\n"
+				       "\n"
+				       "Optional arguments:\n"
+				       "  -h, --help            show this help message and exit\n"
+				       "\n"
+				       "Commands:\n"
+				       "  These commands represent the components of the ADALM2000\n"
+				       "\n"
+				       "    analog-in           control the analogical input component\n"
+				       "    analog-out          control the analogical output component\n"
+				       "    digital             control the digital component\n"
+				       "    power-supply        control the power supply\n";
+
+int main(int argc, char **argv)
+{
+	bool quiet;
+	std::vector<std::pair<std::string, std::string>> output;
+	libm2k::cli::Command *command;
+	try {
+		if (std::string(argv[1]) == "--help" || std::string(argv[1]) == "-h") {
```

Comment:
Should we check if any argument was provided in the command line? 
Maybe just print the "Help" menu if there is no argument? 
Otherwise the user just sees the message of the exception raised by std::string.

Reply:
It is a good idea to print the help message if there are no arguments. The modifications were added in commit 6daf277.